server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -953,6 +967,147 @@ public void testRegisterWithReservedPath() {
         }
     }
 
+    /*
+     * Test that when serverless is disabled, all endpoints are available regardless of ServerlessScope annotations.
+     */
+    public void testApiProtectionDisabled() {
```

Comment:
nit: the naming of these tests are a bit confusing. "disabled" here looks like it relates to api protection. Maybe prefix the enabled/disabled with serverless?